styx-scheduler-service/src/main/java/com/spotify/styx/StateInitializingTrigger.java

```diff
@@ -49,17 +48,15 @@
 
   private final StateFactory stateFactory;
   private final StateManager stateManager;
-  private final Storage storage;
 
-  StateInitializingTrigger(StateFactory stateFactory, StateManager stateManager, Storage storage) {
+  StateInitializingTrigger(StateFactory stateFactory, StateManager stateManager) {
     this.stateFactory = Objects.requireNonNull(stateFactory);
     this.stateManager = Objects.requireNonNull(stateManager);
-    this.storage = Objects.requireNonNull(storage);
   }
 
   @Override
   public CompletionStage<Void> event(Workflow workflow, Trigger trigger, Instant instant) {
-    if (!WorkflowValidator.hasDockerConfiguration(workflow, storage)) {
```

Comment:
We now solely rely on the Workflow object and never check Storage at this stage.
We can see that the Workflow object passed here from various places in the code is created by either reading Storage (from `SchedulerResource` and `TriggerManager`) or just retrieved from WorkflowCache (from `BackfillTriggerManager`). Storage and WorkflowCache _should_ be in sync if data is not manually changed in Datastore.

With this comment I wanted to point out that the `hasDockerConfiguration` check slightly changes now for the `BackfillTriggerManager` case. We should consolidate or remove WorkflowCache (in a different PR of course).

Reply:
agreed, we should rework the workflow cache soon